system/src/commonTest/kotlin/com/splendo/kaluga/system/network/NetworkStateTest.kt

```diff
@@ -43,7 +43,7 @@ class NetworkStateTest : FlowTest<NetworkState, NetworkStateRepo>() {
         assertInitialValue(this)
 
         action {
-            networkStateRepo.onNetworkStateChange(Network.Known.Wifi())
+            (networkStateRepo as MockNetworkStateRepo).simulateNetworkStateChange(Network.Known.Wifi())
```

Comment:
Thanks for the suggestion. I already thought about it but preferred to avoid it for 2 reasons:
- we are not testing a mock, but a real implementation. We use the mock just to simulate input and I prefer to be as explicit as possible about this.
- Having `MockNetworkStateRepo` as a parameter instead of `NetworkStateRepo` would force me to change the signature of `class MockNetworkStateRepo(networkManagerBuilder: BaseNetworkManager.Builder) : NetworkStateRepo(networkManagerBuilder)` to return the mocked version instead. I prefer not to do it, as the mock builder should follow the signature of the regular builder (imho)

So, will merge as it is